manage/upgrading/version_specific_migration/upgrade_to_52.rst

```diff
@@ -5,28 +5,315 @@ Upgrading Plone 5.1 to 5.2
+
+
+New Login
+---------
+
+This is `PLIP 2092 <https://github.com/plone/Products.CMFPlone/issues/2092>`_.
+
+
+For End Users
+~~~~~~~~~~~~~
+
+Nothing changes.
+
+
+For Developers
+~~~~~~~~~~~~~~
+
+Overrides of any templates or Python scripts that dealt with login or logout need to be changed.
+
+The login has moved from skin-based system to browser views.
+You can use ``z3c.jbot`` to override templates and use the component architecture to override the views.
+The main code is now in ``Products.CMFPlone.browser.login.login.LoginForm``.
+
+You can customize the location to which a user will be redirected after login with an adapter.
+Here is an example:
+
+.. code-block:: python
+
+    from plone import api
+    from Products.CMFPlone.interfaces import IRedirectAfterLogin
+    from Products.CMFPlone.utils import safe_unicode
+    from zope.interface import implementer
+
+
+    @implementer(IRedirectAfterLogin)
+    class RedirectAfterLoginAdapter(object):
+
+        def __init__(self, context, request):
+            self.context = context
+            self.request = request
+
+        def __call__(self, came_from=None, is_initial_login=False):
+            if 'Reviewer' in api.user.get_roles():
+                api.portal.show_message(u'Get to work!', self.request)
+                came_from = self.context.portal_url() + '/@@full_review_list'
+            else:
+                user = api.user.get_current()
+                fullname = safe_unicode(user.getProperty('fullname'))
+                api.portal.show_message(u'Nice to see you again, {0}!'.format(fullname), self.request)
+            if not came_from:
+                came_from = self.context.portal_url()
+            return came_from
+
+Then register the adapter through ZCML:
+
+.. code-block:: xml
+
+    <adapter
+        factory="your.addon.adapters.RedirectAfterLoginAdapter"
+        for="OFS.interfaces.ITraversable
+             zope.publisher.interfaces.IRequest"
+        />
+
+This adapter adapts context and request, thus you can modify these according to your needs.
+You can also write similar adapters for ``IInitialLogin`` and ``IForcePasswordChange``.
+
+
+Deprecate Archetypes
+--------------------
+
+This is `PLIP 2390 <https://github.com/plone/Products.CMFPlone/issues/2390>`_.
+
+
+For End Users
+~~~~~~~~~~~~~
+
+Nothing changes.
+
+For Developers
+~~~~~~~~~~~~~~
+
+In Plone 5.2 Archetypes is only available if you run Python 2.7 and if you add it to your dependencies.
+
+You can add it by either adding ``Products.ATContentTypes`` to the list of your add-ons or by using the "extra" ``archetypes`` with the egg ``Plone`` in your buildout:
+
+.. code-block:: ini
+
+    [instance]
+    recipe = plone.recipe.zope2instance
+    eggs =
+        Plone[archetypes]
+        your.addon
+
+Instead of using Archetypes in Plone 5.2, you should consider migrating to Dexterity.
```

Comment:
Personally, I even set this as "note" directive to make that visual more prominent

Reply:
ok

Reply:
I did so